### PR TITLE
Right panel chat style changes for read receipts and optimizations for smaller widths

### PR DESCRIPTION
**Pull request by toger5**
Fixes: https://github.com/vector-im/element-web/issues/19905
This is a first draft for the new styling. 
It tires to accommodate the read receipts and use the space more efficiently when there is little space available like in the right panel chat.
It needs feedback from design!
The two options differ in how aggressive they use horizontal space: 
 - Option 1 really similar to the normal design just a little bit tighter
 - Option 2 uses all the possible area on the left.

## Option 1
option1
## Option 2
option2

## Before:

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://61b75240c441b909d346564a--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by jryans** [comment]
I think we should start with a Design review of this one first, as there are several proposed options. After that's settled, we can continue into the code review.

**Review by gaelledel — COMMENTED** [review]
@toger5 @jryans 

For the full width of the pannel:
Surface out in this order: Emoji , Overflow
Inside the Overflow: Stickers, Voice message, Polls, Location sharing
https://www.figma.com/file/DFBvBKfg2sZw6YrbqOYWK6/Composer?node-id=317%3A12520

If required when reducing the view port size, hide the emoji button inside the overflow.

**Comment by toger5** [comment]
@gaelledel This PR is not intended to modify the composer. Even though that is also interesting. I agree that the emoji picker should be prioritized to the attachment button.
The intention was to only get read recites added, so that the right panel chat is closer to being feature complete.
Feel free to open another issue about the composer. I think it would be best to have this issue only focused on the timeline.

**Comment by gaelledel** [comment]
> @gaelledel This PR is not intended to modify the composer. Even though that is also interesting. I agree that the emoji picker should be prioritized to the attachment button. The intention was to only get read recites added, so that the right panel chat is closer to being feature complete. Feel free to open another issue about the composer. I think it would be best to have this issue only focused on the timeline.

Right gotcha and apologies. In this case, the Proposal you have there works. Will create an issue for the composer (fyi I meant to sync with Jano about exposing emoji for the threads panel too but he's not been available)

**Comment by toger5** [comment]
@gaelledel what approach do you prefer:
 - you and jano choose one of the two options and we mere this rather quickly? (with still having the option to open an issue and improve on it when we have a Figma design) Which of the two option should we go with than?
 - we keep this as a first building block and wait for a proper Figma design which I than (hopefully) can execute based on this.
 
 I really like/need/use the read recites so I am biased to have them in the right panel rather quickly.

**Comment by gaelledel** [comment]
> @gaelledel what approach do you prefer:
> 
> * you and jano choose one of the two options and we mere this rather quickly? (with still having the option to open an issue and improve on it when we have a Figma design) Which of the two option should we go with than?
> * we keep this as a first building block and wait for a proper Figma design which I than (hopefully) can execute based on this.
> 
> I really like/need/use the read recites so I am biased to have them in the right panel rather quickly.


Yes so you can merge this work on read receipts - For the emoji and composer we will see later

**Review by gaelledel — APPROVED** [review]
Let's merge this and we can figure out the composer stuff later

**Comment by toger5** [comment]
@gaelledel : which of the two options from the PR description?
@jryans : I think I should than also add a listener for the read recite setting? (I will request a code review again since this has green light from design now)

File: src/components/views/right_panel/TimelineCard.tsx
Comment (toger5) [inline src/components/views/right_panel/TimelineCard.tsx:77]: I maybe should we add a listener for the setting?

Comment (jryans) [reply src/components/views/right_panel/TimelineCard.tsx:77]: It's nice to use listeners for quick update to settings changes, but we're also quite inconsistent about it throughout, so whatever seems best to you is okay.

**Comment by toger5** [comment]
Settings updates are tested now. I prefer Option 1. So I am going to merge this this evening if there are not further objections (I'll keep the comment for the other option in case we want to change that later. If this is bad practice I will remove them before merging)

**Comment by jryans** [comment]
> So I am going to merge this this evening if there are not further objections

Well, just to clarify process-wise, no one has approved the code yet, so that's needed first. 😉 Anyway, I will review it now.

File: res/css/views/right_panel/_TimelineCard.scss
Comment (jryans) [inline res/css/views/right_panel/_TimelineCard.scss:99]: No need to keep alternatives like this in the final merge.

```suggestion
```

File: src/components/structures/RightPanel.tsx
Comment (jryans) [inline src/components/structures/RightPanel.tsx:341]: What's the reason for exposing this as a prop when it seems to only have one value? Will it be set to something different elsewhere in the future...?

**Review by jryans — CHANGES_REQUESTED** [review]
Almost ready, just a small question to resolve. 😄

File: src/components/structures/RightPanel.tsx
Comment (toger5) [reply src/components/structures/RightPanel.tsx:341]: yes exactly, the idea is, that the threads team can use the TimelineCard for the right panel threads, currently they are using different styling so to make the switch to the TimelineCard it needs to be possible to set a custom class.

**Comment by toger5** [comment]
> Well, just to clarify process-wise, no one has approved the code yet, so that's needed first. wink Anyway, I will review it now.

sorry you are right. There was no code review yet. Thanks taking the time to review now!

**Review by jryans — APPROVED** [review]
Thanks, looks good to me! 😄

File: res/css/views/right_panel/_TimelineCard.scss
Comment (MadLittleMods) [inline res/css/views/right_panel/_TimelineCard.scss:33]: From [@toger5's demo](https://matrix.to/#/!kxwQeJPhRigXSZrHqf:matrix.org/$Mwb4wKGLgGdimhg6-DcpIkZCLHgOWWQXjP7HvBf3pTU?via=matrix.org&via=element.io&via=vector.modular.im) of maximized widgets where it showed the right-panel chat style, I was surprised that it looked a lot more native and refined compared to the thread view. This probably comes down to the differences from the normal timeline view because we're changing some the pieces around like the timestamp position in the thread view. I understand that the right panel chat style and the thread view are different.

Right panel chat style that feels nice to me:


The unpolished feeling may apply to this PR iteration of the right-panel chat where things are getting condensed like the thread view but it's hard to judge because I haven't actually used it in the app like threads and it's nice to be able to judge the balance against the whole app and timeline view.

As @toger5 notes, some adaptations probably need to be made for the smaller layout,

> We definitely have to make changes to the style. Just to get read-receipts not glitching/overlapping with the messages. The question is, if the style changes are too aggressive. I see both sides. It looking more like the main timeline (more native) when changing as a little as possible and adapting it to the right panel so the small area is used as efficient as possible.
>
> *-- @toger5, https://matrix.to/#/!tCaHWuHHvOHjWyqapP:matrix.org/$UibGBNcTpdbe-0zdRn1Ktp8_yHWYh9wL6HlfxJ3t-tc?via=matrix.org&via=element.io&via=uhoreg.ca*


---

As a tiny point of reference, Slack and Discord are able to maintain the same layout between the main message view and threads but they have a different timestamp position and less UI (don't need to contend with read-receipts).

**Pull request merged**
